File: refactor_colors.dart
import 'dart:io';

void main() {
  final dir = Directory('lib');
  final files = dir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart'));

  for (final file in files) {
    if (file.path.contains('app_theme.dart') || file.path.contains('app_colors.dart')) continue;
    
    var content = file.readAsStringSync();
    
    if (content.contains('AppColors.')) {
      content = content.replaceAll('AppColors.paper', 'context.colors.paper');
      content = content.replaceAll('AppColors.ink2', 'context.colors.ink2');
      content = content.replaceAll('AppColors.ink3', 'context.colors.ink3');
      content = content.replaceAll('AppColors.ink4', 'context.colors.ink4');
      content = content.replaceAll('AppColors.ink', 'context.colors.ink');
      content = content.replaceAll('AppColors.rule2', 'context.colors.rule2');
      content = content.replaceAll('AppColors.rule', 'context.colors.rule');
      content = content.replaceAll('AppColors.wash', 'context.colors.wash');
      content = content.replaceAll('AppColors.accent', 'context.colors.accent');
      content = content.replaceAll('AppColors.dangerRule', 'context.colors.dangerRule');
      content = content.replaceAll('AppColors.danger', 'context.colors.danger');
      
      // Also add the extension import if not present
      if (!content.contains('core/theme/app_theme.dart')) {
        // find a good place to import
        final importsEnd = content.lastIndexOf('import ');
        if (importsEnd != -1) {
          final nextLine = content.indexOf('\n', importsEnd);
          
          // figure out relative path to core/theme/app_theme.dart
          // Count directory depth
          final depth = file.path.split(Platform.pathSeparator).length - 2; // -1 for 'lib', -1 for filename
          final relativePrefix = depth == 0 ? '' : List.filled(depth, '../').join('');
          final importStmt = "import '${relativePrefix}core/theme/app_theme.dart';\n";
          
          content = content.substring(0, nextLine + 1) + importStmt + content.substring(nextLine + 1);
        }
      }
      
      file.writeAsStringSync(content);
      print('Updated \${file.path}');
    }
  }
}

File: refactor_colors.py
import os

def process_file(filepath):
    if 'app_theme.dart' in filepath or 'app_colors.dart' in filepath:
        return
        
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
        
    if 'AppColors.' in content:
        content = content.replace('AppColors.paper', 'context.colors.paper')
        content = content.replace('AppColors.ink2', 'context.colors.ink2')
        content = content.replace('AppColors.ink3', 'context.colors.ink3')
        content = content.replace('AppColors.ink4', 'context.colors.ink4')
        content = content.replace('AppColors.ink', 'context.colors.ink')
        content = content.replace('AppColors.rule2', 'context.colors.rule2')
        content = content.replace('AppColors.rule', 'context.colors.rule')
        content = content.replace('AppColors.wash', 'context.colors.wash')
        content = content.replace('AppColors.accent', 'context.colors.accent')
        content = content.replace('AppColors.dangerRule', 'context.colors.dangerRule')
        content = content.replace('AppColors.danger', 'context.colors.danger')
        
        if 'core/theme/app_theme.dart' not in content:
            last_import = content.rfind('import ')
            if last_import != -1:
                next_line = content.find('\n', last_import)
                
                parts = filepath.replace('\\', '/').split('/')
                depth = len(parts) - 2
                prefix = '../' * depth if depth > 0 else ''
                import_stmt = f"\nimport '{prefix}core/theme/app_theme.dart';"
                
                content = content[:next_line] + import_stmt + content[next_line:]
                
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f'Updated {filepath}')

for root, _, files in os.walk('lib'):
    for file in files:
        if file.endswith('.dart'):
            process_file(os.path.join(root, file))
